lib/ansible/modules/network/cli/cli_config.py

```diff
@@ -0,0 +1,284 @@
+            not capabilities['device_operations']['supports_diff_ignore_lines']):
+        _handle_severity(module, msg='diff_ignore_lines is not supported on this platform')
+
+
+def run(module, connection, candidate, running):
+    result = {}
+
+    replace = module.params['replace']
+    rollback = module.params['rollback']
+    commit_comment = module.params['commit_comment']
+    multiline_delimiter = module.params['multiline_delimiter']
+    diff_replace = module.params['diff_replace']
+    diff_match = module.params['diff_match']
+    diff_ignore_lines = module.params['diff_ignore_lines']
+
+    commit = not module.check_mode
+
+    kwargs = {'candidate': candidate, 'running': running}
+    if diff_match:
+        kwargs.update({'match': diff_match})
+    if diff_replace:
+        kwargs.update({'replace': diff_replace})
+    if diff_ignore_lines:
+        kwargs.update({'diff_ignore_lines': diff_ignore_lines})
+
+    diff_response = connection.get_diff(**kwargs)
+
+    config_diff = diff_response.get('config_diff')
+    banner_diff = diff_response.get('banner_diff')
+
+    if config_diff:
+        if isinstance(config_diff, list):
+            candidate = config_diff
+        else:
+            candidate = config_diff.splitlines()
+
+        kwargs = {'candidate': candidate, 'commit': commit, 'replace': replace,
+                  'comment': commit_comment}
+        try:
+            resp = connection.edit_config(**kwargs)
+            if 'diff' in resp:
+                if resp['diff']:
+                    result['commands'] = resp['diff']
+            else:
+                result['commands'] = candidate
+        except ConnectionError as exc:
+            msg = to_text(exc)
+            if 'check mode is not supported' in msg:
+                _handle_severity(module, msg=msg)
+            else:
+                module.fail_json(msg=msg)
+
+    if banner_diff:
+        candidate = banner_diff
+        result['banners'] = candidate
+
+        kwargs = {'candidate': candidate, 'commit': commit}
+        if multiline_delimiter:
+            kwargs.update({'multiline_delimiter': multiline_delimiter})
+        connection.edit_banner(**kwargs)
+
+    if any(key in result for key in ('commands', 'banners')):
+        result['changed'] = True
+
+    return result
+
+
+def main():
+    """main entry point for execution
+    """
+    argument_spec = dict(
+        config=dict(required=True, type='str'),
+        commit=dict(type='bool'),
+        replace=dict(default=False, type='bool'),
+        rollback=dict(type='int'),
+        commit_comment=dict(type='str'),
+        defaults=dict(default=False, type='bool'),
+        multiline_delimiter=dict(type='str'),
+        diff_replace=dict(choices=['line', 'block', 'config']),
+        diff_match=dict(choices=['line', 'strict', 'exact', 'none']),
+        diff_ignore_lines=dict(type='list'),
+        severity=dict(default='error', choices=['error', 'warning'])
+    )
```

Comment:
You can also add option the enable/disable logging of command and it's response in output using [enable_response_logging](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/cliconf/__init__.py#L159) and [disable_response_logging](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/cliconf/__init__.py#L163) api